gulpfile.js

```diff
@@ -85,6 +86,84 @@ function createReloadServer() {
   });
 }
 
+function writeServiceWorkerFile() {
+  let path = require('path');
+  let rootDir = 'dist';
+
+  let dynamicUrlToDependencies = {};
+  let partialTemplateFiles = ['head-meta.html', 'site-nav.html']
+    .map(file => path.join(rootDir, 'templates', file));
+
+  ['about', 'index', 'shell'].forEach(htmlFile => {
+    let dynamicUrl = '/' + htmlFile;
+    let dependencies = partialTemplateFiles.concat(`${rootDir}/${htmlFile}.html`);
+    dynamicUrlToDependencies[dynamicUrl] = dependencies;
+    // Treat / like /index
+    if (htmlFile === 'index') {
+      dynamicUrlToDependencies['/'] = dependencies;
+    }
+  });
+
+  let config = {
+    cacheId: 'polymerproject',
+    staticFileGlobs: [
+      `${rootDir}/images/logos/p-logo.png`,
+      `${rootDir}/elements/**`,
+      `${rootDir}/js/*.js`,
+      `${rootDir}/css/*.css`,
+      `${rootDir}/bower_components/**/webcomponents-lite.min.js`,
+      `${rootDir}/bower_components/highlight/highlight.js`,
+    ],
+    dynamicUrlToDependencies: dynamicUrlToDependencies,
+    navigateFallback: '/shell',
```

Comment:
@jeffposnick Just had the chance to catch up on your earlier comments in this PR (before I was mentioned in this PR). One of the reasons why the docs site doesn't follow the app shell structure is that most of the changing content on the site are the docs themselves and not the layout/structure/code. The docs are rich content HTML (e.g. headings, paragraphs, code snippets, inline iframe demos), and the structure can vary from page to page. So the best way to encode docs is in HTML and not HTML embedded in JSON (or something) that is fetched and processed by the shell. The structure of a page is primarily defined by the definition of `<pw-shell>`, so if we were to change the layout of the site, only the precached file of `<pw-shell>` would change and the content pages wouldn't need to be updated.

Another reason why the app shell pattern may not be appropriate is that we need all the content of the site to be crawlable. At time of writing, Googlebot runs JS, but many other search engines do not so if content isn't in HTML, it won't be detected (e.g. searching for "polymer square brackets data binding" - there's no page titled that or has that in meta description, but it's in the content).

I like the idea of disabling `navigateFallback` and a custom runtime caching for the docs pages (e.g. /2.0/toolbox/, /blog/, etc.), something akin to network-first with a fallback on network-miss and cache-miss. (I prefer network-first over stale-and-revalidate because we shouldn't show stale docs content if we don't have to). I'm gonna try to see if I can make this work by adding a new SW JS file and using the `importScripts` config.